src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -343,15 +343,28 @@ TypeAnnotations BuildTypeAnnotations (TypeDefinition type)
 							ScanMethodBodyForFieldAccess (setMethod.Body, write: true, out backingFieldFromSetter);
 						}
 
-						if (annotatedMethods.Any (a => a.Method == setMethod)) {
+						MethodAnnotations? setterAnnotation = null;
+						foreach (var annotatedMethod in annotatedMethods) {
+							if (annotatedMethod.Method == setMethod)
+								setterAnnotation = annotatedMethod;
+						}
+
+						// If 'value' parameter is annotated, then warn. Other parameters can be annotated for indexable properties
+						if (setterAnnotation?.ParameterAnnotations?[^1] is not (null or DynamicallyAccessedMemberTypes.None)) {
 							_context.LogWarning (setMethod, DiagnosticId.DynamicallyAccessedMembersConflictsBetweenPropertyAndAccessor, property.GetDisplayName (), setMethod.GetDisplayName ());
 						} else {
 							int offset = setMethod.HasImplicitThis () ? 1 : 0;
-							if (setMethod.Parameters.Count > 0) {
-								DynamicallyAccessedMemberTypes[] paramAnnotations = new DynamicallyAccessedMemberTypes[setMethod.Parameters.Count + offset];
-								paramAnnotations[paramAnnotations.Length - 1] = annotation;
-								annotatedMethods.Add (new MethodAnnotations (setMethod, paramAnnotations, DynamicallyAccessedMemberTypes.None, null));
-							}
+							if (setterAnnotation is not null)
```

Comment:
I think we could, but that makes me wonder if we should actually make `MethodAnnotations.ParameterAnnotations` an `ImmutableArray<DynamicallyAccessedMembers>` instead of just a mutable array. The fields are already readonly, so I think the intent is that they are immutable data structures.